pkgs/by-name/de/deadlock-modmanager/package.nix

```
@@ -0,0 +1,155 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  cargo-tauri,
+  nodejs,
+  pnpm_9,
+  pkg-config,
+  wrapGAppsHook3,
+  desktop-file-utils,
+  webkitgtk_4_1,
+  cairo,
+  gdk-pixbuf,
+  glib,
+  glib-networking,
+  gtk3,
+  libsoup_3,
+  pango,
+  openssl,
+  bzip2,
+  gst_all_1,
+  makeDesktopItem,
+  fontconfig,
+  nix-update-script,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "deadlock-modmanager";
+  version = "0.9.2";
+
+  src = fetchFromGitHub {
+    owner = "Stormix";
+    repo = "deadlock-modmanager";
+    tag = "v${version}";
+    hash = "sha256-meCwpdnRYwNiOB8QzdhDC24CtRiMClimCDlv0EvHHgI=";
+  };
+
+  cargoRoot = "apps/desktop";
+
+  cargoHash = "sha256-cLdsR9ZCkUKZQDAxsvZpau7c9LxeLt4kUI21ObiY7dA=";
+
+  nativeBuildInputs = [
+    rustPlatform.cargoSetupHook
+    cargo-tauri
+    nodejs
+    pnpm_9.configHook
+    pkg-config
+    wrapGAppsHook3
+  ];
+
+  buildInputs = [
+    webkitgtk_4_1
+    cairo
+    gdk-pixbuf
+    glib
+    glib-networking
+    gtk3
+    libsoup_3
+    pango
+    openssl
+    bzip2
+    desktop-file-utils
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-bad
+  ];
+
+  pnpmRoot = ".";
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit
+      pname
+      version
+      src
+      ;
+    fetcherVersion = 2;
+    sourceRoot = "source";
+    hash = "sha256-ZnVUQjPXphwk7J5jd04WQdUYQF0t5veqRVr/m4guuEU=";
+  };
+
+  patches = [
+    ./no-updater-artifacts.patch
+  ];
+
+  VITE_API_URL = "https://api.deadlockmods.app";
+
+  buildPhase = ''
+    runHook preBuild
+
+    cd apps/desktop
+    export CARGO_TARGET_DIR=target
+
+    cargo tauri build --no-bundle -- --frozen
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 src-tauri/target/release/deadlock-mod-manager $out/bin/deadlock-modmanager
+
+    install -Dm644 src-tauri/icons/32x32.png $out/share/icons/hicolor/32x32/apps/deadlock-modmanager.png
+    install -Dm644 src-tauri/icons/128x128.png $out/share/icons/hicolor/128x128/apps/deadlock-modmanager.png
+    install -Dm644 src-tauri/icons/128x128@2x.png $out/share/icons/hicolor/256x256/apps/deadlock-modmanager.png
+
+    mkdir -p $out/share/deadlock-modmanager
+    cp -r dist $out/share/deadlock-modmanager/
+
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --set FONTCONFIG_FILE "${fontconfig.out}/etc/fonts/fonts.conf"
+      --set TAURI_DIST_DIR "$out/share/deadlock-modmanager/dist"
+      --set WEBKIT_DISABLE_COMPOSITING_MODE 1
+      --set WEBKIT_DISABLE_DMABUF_RENDERER 1
+      --set DISABLE_UPDATE_DESKTOP_DATABASE 1
+      --prefix PATH : ${lib.makeBinPath [ desktop-file-utils ]}
+    )
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      desktopName = "Deadlock Mod Manager";
```

```suggestion
      desktopName = "deadlock-modmanager";
```

Probably safer since I'm not sure if apps tend to choke on spaces in desktop filenames. App launchers use `name` not `desktopName` anyway to display the app names.